module-ama/src/main/kotlin/org/quiltmc/community/cozy/modules/ama/_Utils.kt

```diff
@@ -0,0 +1,385 @@
+											style = ButtonStyle.Primary
+
+											action {
+												val claimer = event.interaction.user
+												answerQueueMessage?.edit {
+													embed {
+														questionEmbed(
+															interactionUser,
+															question,
+															QuestionStatusFlag.CLAIMED,
+															claimedOrSkippedBy = event.interaction.user
+														)
+													}
+													components { removeAll() }
+													components {
+														answeringButtons(
+															claimer,
+															liveChatChannel,
+															interactionUser,
+															question,
+															answerQueueMessage
+														)
+													}
+												}
+											}
+										}
+
+										ephemeralButton {
+											label = "Skip"
+											style = ButtonStyle.Secondary
+
+											action {
+												answerQueueMessage?.edit {
+													embed {
+														questionEmbed(
+															interactionUser,
+															question,
+															QuestionStatusFlag.SKIPPED,
+															claimedOrSkippedBy = event.interaction.user
+														)
+													}
+													components { removeAll() }
+												}
+											}
+										}
+									}
+								}
+								flaggedMessage?.edit {
+									embed {
+										questionEmbed(
+											interactionUser,
+											question,
+											QuestionStatusFlag.ACCEPTED
+										)
+									}
+									components { removeAll() }
+								}
+								embedMessage.edit {
+									embed {
+										questionEmbed(
+											interactionUser,
+											question,
+											QuestionStatusFlag.ACCEPTED
+										)
+									}
+									components { removeAll() }
+								}
+							}
+						}
+
+						ephemeralButton {
+							label = "Deny"
+							style = ButtonStyle.Danger
+
+							action {
+								flaggedMessage?.edit {
+									embed {
+										questionEmbed(interactionUser, question, QuestionStatusFlag.DENIED)
+									}
+									components { removeAll() }
+								}
+								embedMessage.edit {
+									embed {
+										questionEmbed(
+											interactionUser,
+											question,
+											QuestionStatusFlag.DENIED
+										)
+									}
+									components { removeAll() }
+								}
+							}
+						}
+					}
+				}
+				embedMessage.edit {
+					embed {
+						questionEmbed(
+							interactionUser,
+							question,
+							QuestionStatusFlag.FLAGGED,
+							flaggedBy = event.interaction.user
+						)
+					}
+					components { removeAll() }
+				}
+			}
+		}
+	}
+}
+
+public suspend inline fun ComponentContainer.answeringButtons(
+	claimer: User,
+	liveChatChannel: GuildMessageChannel?,
+	interactionUser: User,
+	question: String?,
+	answerQueueMessage: Message?
+) {
+	ephemeralButton {
+		label = "Stage"
+		style = ButtonStyle.Success
+
+		check {
+			if (event.interaction.user != claimer) {
+				fail("You did not claim this question! ${claimer.tag} did")
+			} else {
+				pass()
```

Comment:
I'm pretty sure it passes by default, so this is unnecessary

Reply:
This is true